checker_tool/suites/eus_installer/general.sh

```diff
@@ -1,23 +1,198 @@
+        cs_operator_channel=$(oc get subscription.operators.coreos.com ibm-common-service-operator -n openshift-operators --ignore-not-found -o yaml | yq ".spec.channel")
+    fi
+    # if not found cert-manager sub in operator namespace, check cs-control namespace
+    if [[ "$cert_manager_channel" == "" ]]; then
+        cert_manager_channel=$(oc get subscription.operators.coreos.com ibm-cert-manager-operator -n cs-control --ignore-not-found -o yaml | yq ".spec.channel")
+    fi
+
+    if [[ "$iam_operator_channel" == "stable-v1" ]] && [[ "$odlm_channel" == "stable-v1" ]] && [[ "$cs_operator_channel" == "stable-v1" ]] && [[ "$cert_manager_channel" == "stable-v1" ]]; then
+        success "cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel stable-v1 for EUS"
+        append_check "eus_installer" "check subscription channel" "ok" "cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel stable-v1 for EUS" ""
+    elif [[ "$iam_operator_channel" == "v3" ]] && [[ "$odlm_channel" == "v3" ]] && [[ "$cs_operator_channel" == "v3" ]] && [[ "$cert_manager_channel" == "v3" ]]; then
+        success " cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel v3 for LTSR"
+        append_check "eus_installer" "check subscription channel" "ok" "cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel v3 for LTSR" ""
+    elif [[ "$iam_operator_channel" == "v3.23" ]] && [[ "$odlm_channel" == "v3.23" ]] && [[ "$cs_operator_channel" == "v3.23" ]] && [[ "$cert_manager_channel" == "v3.23" ]]; then
+        success " cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel v3 for LTSR"
+        append_check "eus_installer" "check subscription channel" "ok" "cert-manager, IAM, ODLM and CS-Operator subscriptions are all in correct channel v3.23 for LTSR" ""
+    else
+        error "operators may not in correct channel, please check"
+        append_check "eus_installer" "check subscription channel" "failed" "cert-manager:${cert_manager_channel}, IAM:${ibm_iam_operator_channel}, ODLM:${odlm_channel} and CS-Operator:${cs_operator_channel} subscriptions may not in correct channel" ""
+        msg "ibm-common-service-operator channel: ${cs_operator_channel}"
+        msg "ibm-iam-operator channel: ${ibm_iam_operator_channel}"
+        msg "operand-deployment-lifecycle-manager channel: ${odlm_channel}"
+        msg "ibm-cert-manager-operator channel: ${cert_manager_channel}"
+    fi
+
+}
+
+# check if all the CSV in ibm-common-services namespace are in succeeded status
+function check_csv() { 
+    title "Checking CSV status in namespace: ${OPERATOR_NS}"
+    oc get csv -n ${OPERATOR_NS}
+    failed_csv=$(oc get clusterserviceversion.operators.coreos.com -n ${OPERATOR_NS} -o=jsonpath='{.items[?(@.status.phase != "Succeeded")].metadata.name}')
+    if [[ "X$failed_csv" != "X" ]]; then
+        for csv in $(echo $failed_csv);
```

Comment:
```suggestion
        for csv in $(echo $failed_csv)
```

No need for semi-colon if the `do` keyword is on the next line